EOLib/PacketHandlers/Players/PlayersAgreeHandler.cs

```diff
@@ -33,37 +34,22 @@ public class PlayersAgreeHandler : InGameOnlyPacketHandler
                                      ICurrentMapStateRepository mapStateRepository,
                                      ICharacterFromPacketFactory characterFromPacketFactory,
                                      IEIFFileProvider eifFileProvider,
-                                     IEnumerable<IEffectNotifier> effectNotifiers)
+                                     IEnumerable<IEffectNotifier> effectNotifiers,
+                                     IPacketTranslator<PlayersAgreeData> playersAgreeTranslator)
             : base(playerInfoProvider)
         {
             _characterRepository = characterRepository;
             _mapStateRepository = mapStateRepository;
             _characterFromPacketFactory = characterFromPacketFactory;
             _eifFileProvider = eifFileProvider;
             _effectNotifiers = effectNotifiers;
+            _playersAgreeTranslator = playersAgreeTranslator;
         }
 
         public override bool HandlePacket(IPacket packet)
         {
-            if (packet.ReadByte() != 255)
-                throw new MalformedPacketException("Missing 255 header byte for player enter map handler", packet);
-
-            var character = _characterFromPacketFactory.CreateCharacter(packet);
-
-            if (packet.PeekByte() != 255) // next byte was the warp animation: sent on Map::Enter in eoserv
-            {
-                var anim = (WarpAnimation)packet.ReadChar();
-
-                foreach (var notifier in _effectNotifiers)
-                    notifier.NotifyWarpEnterEffect(character.ID, anim);
-            }
-
-            if (packet.ReadByte() != 255)
-                throw new MalformedPacketException("Missing 255 byte after the warp animation for player enter map handler", packet);
-
-            // 0 for NPC, 1 for player. In eoserv it is never 0.
-            if (packet.ReadChar() != 1)
-                throw new MalformedPacketException("Missing '1' char after warp animation for player enter map handler. Are you using a non-standard version of EOSERV?", packet);
+            var data = _playersAgreeTranslator.TranslatePacket(packet);
```

Comment:
This breaks eoserv compatibility. The PLAYERS_AGREE packet that eoserv sends is just...wrong. But because of the way the chunked reader works, it handles the wrong data correctly. EndlessClient won't have this functionality until eolib-dotnet is done.

To repro, try warping with another character to the map the main character is on.

Reply:
For more information on the gnarly `PLAYERS_AGREE` data sent by the official server (and blindly mimicked by EOSERV), see the eo-protocol docs.

https://github.com/Cirras/eo-protocol/blob/master/docs/chunks.md#a-practical-example

Reply:
@ethanmoffat also realized the solution still isn't correct because `_characterFromPacketFactory.CreateCharacter` doesn't check for the WarpAnimation byte.

Idk where the proper place to put this would be. On Character itself or Render properties?

Reply:
I'm really not sure of a good solution for this. I'm not devoting a lot of thinking time to it because eolib-dotnet will remove the concept of "packet translators" from EndlessClient.